itests/fevm_test.go

```diff
@@ -1289,3 +1290,80 @@ func TestEthGetTransactionCount(t *testing.T) {
 	require.NoError(t, err)
 	require.Zero(t, contractNonceAfterDestroy)
 }
+
+func TestEthGetBlockByNumber(t *testing.T) {
+	blockTime := 100 * time.Millisecond
+	client, _, ens := kit.EnsembleMinimal(t, kit.MockProofs(), kit.ThroughRPC())
+
+	bms := ens.InterconnectAll().BeginMining(blockTime)
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+
+	// Create a new Ethereum account
+	_, ethAddr, filAddr := client.EVM().NewAccount()
+	// Send some funds to the f410 address
+	kit.SendFunds(ctx, t, client, filAddr, types.FromFil(10))
+
+	// Test getting the latest block
+	latest, err := client.EthBlockNumber(ctx)
+	require.NoError(t, err)
+	latestBlock, err := client.EthGetBlockByNumber(ctx, "latest", true)
+	require.NoError(t, err)
+	require.NotNil(t, latestBlock)
+	require.Equal(t, latest, latestBlock.Number)
+
```

Comment:
@virajbhartiya The above is racy. What if a new block is created between the calls to `EthBlockNumber` and `EthGetBlockByNumbe` by the test miner ? These calls will return different blocks then.

I think just asking for the "latest" to `EthGetBlockByNumber` and making sure it's not nil and no error is fine.

Reply:
Oh yes I didn't take into consideration about blocks being mined between calls. I'll fix this!